src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,201 @@
+
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id ", e);
+        }
+        for (Car car: cars) {
+            car.setDriverList(getDriversCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars "
+                + "SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in driversDB.", e);
+        }
+        deleteDriverByCarId(car.getId());
+        insertDriversForCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete driver with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, model, manufacturer_id, m.id "
+                + "AS manufacture_id, m.name, m.country"
+                + "FROM cars c JOIN cars_drivers ON c.id = cars_drivers.car_id"
+                + " JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + "WHERE  cars_drivers.driver_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1,driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars by driver id " + driverId, e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        return new Car(resultSet.getObject("car_id", Long.class),
+                resultSet.getString("model"), getManufacturer(resultSet));
```

Comment:
It wil be better to format such code in that way:
```suggestion
        return new Car(
                resultSet.getObject("car_id", Long.class),
                resultSet.getString("model"), 
                getManufacturer(resultSet));
```